reports/serializers.py

```diff
@@ -38,9 +38,113 @@ def to_internal_value(self, data):
         return [data["longitude"], data["latitude"]]
 
 
+class OCDIDField(serializers.Field):
+    """A serializer for Open Civic Data ID"""
+
+    translations = {
+        "maa": "country",
+        "peruspiiri": "district",
+        "osa-alue": "sub_district",
+        "kaupunginosa": "neighborhood",
+        "kunta": "municipality",
+    }
+
+    def to_representation(self, value: list) -> Optional[dict]:
+        """OCD id paths to a more detailed dict
+
+        Args:
+            value (list): A list of OCD ids, e.g.
+            ```
+            [
+                "ocd-division/country:fi/kunta:helsinki/osa-alue:keski-pasila",
+                "ocd-division/country:fi/kunta:helsinki/kaupunginosa:pasila",
+                "ocd-division/country:fi/kunta:helsinki/peruspiiri:pasila",
+                "ocd-division/country:fi/kunta:helsinki",
+            ]
+            ```
+
+        Returns:
+            dict: the original OCD ids list in the key "ocd-ids" and
+            all the key-value -pairs of OCD id path, e.g.
+            ```
+            "ocd_ids": [
+                "ocd-division/country:fi/kunta:helsinki/osa-alue:keski-pasila",
+                "ocd-division/country:fi/kunta:helsinki/kaupunginosa:pasila",
+                "ocd-division/country:fi/kunta:helsinki/peruspiiri:pasila",
+                "ocd-division/country:fi/kunta:helsinki",
+            ],
+            "country": "Fi",
+            "municipality": "Helsinki",
+            "sub_district": "Keski-pasila",
+            "neighborhood": "Pasila",
+            "district": "Pasila",
+            ```
+        """
+        if value:
+            joined = "/".join(value).replace("ocd-division/", "")
+            pairs = [tuple(pair.split(":")) for pair in set(joined.split("/"))]
+            return {
+                "ocd_ids": value,
+                **{
+                    self.translations[key]
+                    if key in self.translations
+                    else key: value.capitalize()
+                    for (key, value) in pairs
+                },
+            }
+        return None
+
+    def to_internal_value(self, data: dict) -> list:
```

Comment:
I guess we don't actually need to support deserialization since the reporting API is not taking any input, but since this is already implemented and tested there's no harm in having it. :+1: